packages/stager/lib/src/scene.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+/// The central class of Stager, used to demonstrate a single piece of UI.
+///
+/// Use [setUp] to configure dependencies and [build] to create the Widget you
+/// would like to develop or demo. You may find yourself wanting to use the
+/// same [build] implementation with different [setUp] implementations (or vice
+/// versa). In these cases, you can create a base Scene class defines the code
+/// you wish to share and extend that. The following example defines a shared
+/// base Scene that is extended to configure the Api dependency in different
+/// ways while sharing the same [build] implementation.
+///
+/// ```
+/// @GenerateMocks([Api])
+/// abstract class BasePostsListScene extends Scene {
+///   late MockApi mockApi;
+///
+///   @override
+///   Widget build() {
+///     return EnvironmentAwareApp(
+///       home: Provider<Api>.value(
+///         value: mockApi,
+///         child: const PostsList(),
+///       ),
+///     );
+///   }
+///
+///   @override
+///   Future<void> setUp() async {
+///     mockApi = MockApi();
+///   }
+/// }
+///
+/// class EmptyListScene extends BasePostsListScene {
+///   @override
+///   String get title => 'Empty List';
+///
+///   @override
+///   Future<void> setUp() async {
+///     await super.setUp();
+///     when(mockApi.fetchPosts()).thenAnswer((_) async => []);
+///   }
+/// }
+///
+/// class WithPostsScene extends BasePostsListScene {
+///   @override
+///   String get title => 'With Posts';
+///
+///   @override
+///   Future<void> setUp() async {
+///     await super.setUp();
+///     when(mockApi.fetchPosts()).thenAnswer((_) async => Post.fakePosts);
+///   }
+/// }
+/// ```
+///
+abstract class Scene {
```

Comment:
This clashes with a core Flutter class: https://api.flutter.dev/flutter/dart-ui/Scene-class.html

Reply:
☹️ 

Is the only solution to rename? I imagine that a file using Stager would probably not also be using dart:ui, but I suppose there's probably someone out there who would want that. Is it awful to suggest using `import as` to disambiguate? i.e., `import 'package:stager/stager.dart' as stager;` and `class MyScene extends stager.Scene` (or `import 'dart:ui' as ui;` and `ui.Scene`)?

Reply:
Renamed to `StagerScene`

Reply:
same question. Are people really going to have these imports (configuration of an app-under-test and low level dart:ui access) in the same file? The framework itself already conflicts with dart:ui classes a bunch already and import 'dart:ui' as ui is already a common prior art.